boot/bootutil/src/image_validate.c

```diff
@@ -373,6 +375,35 @@ bootutil_get_img_security_cnt(struct image_header *hdr,
     return 0;
 }
 
+#if defined(MCUBOOT_SIGN_PURE)
+/* Returns:
+ *  0 -- found
+ *  1 -- not found
+ * -1 -- failed for some reason
+ *
+ * Value of TLV does not matter, presence decides.
+ */
+static int bootutil_check_for_pure(const struct image_header *hdr,
+                                   const struct flash_area *fap)
+{
+    struct image_tlv_iter it;
+    uint32_t off;
+    uint16_t len;
+    int32_t rc;
+
+    rc = bootutil_tlv_iter_begin(&it, hdr, fap, IMAGE_TLV_SIG_PURE, false);
```

Comment:
security vulnerability: this can reside in protected or unprotected TLV section, so someone can manipulate an image to either have or not have this from an update that has the inverse - can that cause problems? And why is this not forced to protected area?

Reply:
This only indicates that image is build with pure signature, if you turn this off the MCUboot that was build for pure will just not validate image as it expects "pure image"; if you pass image with this to something that does not understand pure, the chances of validation are slim, because it will just try to validate sha against pure signature.
The real reason for this is to be able to distinguish between images, because otherwise you can not tell the signatures apart, when trying to solve issues.